src/expr/src/proto/scalar/func.proto

```diff
@@ -369,3 +369,154 @@ message ProtoUnaryFunc {
         google.protobuf.Empty mz_type_name = 214;
     }
 }
+
+message ProtoBinaryFunc {
+    oneof kind {
+        google.protobuf.Empty and = 1;
```

Comment:
Is this in order of decreasing popularity? If not, should we reorder this slightly so that the first 15 slots go to the most popular functions?

Reply:
I'm happy to do so, but do we have an estimate what the 15 most popular functions are?

Reply:
Do a survey in Slack a la what was done for `ScalarType`? https://materializeinc.slack.com/archives/CM7ATT65S/p1649665254022229

As for what I think, based on looking at chbench.slt/tpch.slt/the occasional user DDL, I think the following are definitely amongst most popular binary functions:
```
Eq
Neq
And
Or
Lt
Lte
Gt
Gte
```

+-*/ are also popular (*/ more so than +-, though we'd need to determine which types are most commonly used with multiplication and division. My inclination is divfloat64, mulnumeric, subnumeric, add_time_interval.

LIKE might also be relatively popular.

Reply:
Recording this as a follow-up task.